cmd/lakectl/cmd/annotate.go

```diff
@@ -0,0 +1,53 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+const annotateTemplate = `{{ range $val := .Commits}}
+{{ $.Object|ljust 10}} {{ $val.Committer }} {{ $val.Id | printf "%.16s"|ljust 20 }} {{ $val.Message |ljust 30 }} {{ $val.CreationDate|date }}{{ end -}}
+`
+
+var annotateCmd = &cobra.Command{
+	Use:   "annotate <path uri>",
+	Short: "Show who created the latest commit",
+	Long:  "Show who created the latest commit to an object or objects in a give path",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, c_args []string) {
+		pathURI := MustParsePathURI("path", c_args[0])
+		client := getClient()
+		amount := 1
+		pfx := api.PaginationPrefix(*pathURI.Path)
+		res, err := client.ListObjectsWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, &api.ListObjectsParams{Prefix: &pfx})
+		DieOnResponseError(res, err)
+		objList := res.JSON200.Results
+		logCommitsParams := &api.LogCommitsParams{
+			Amount: api.PaginationAmountPtr(amount),
+		}
+		pagination := api.Pagination{HasMore: true}
+		for _, obj := range objList {
+			res, err := client.LogCommitsWithResponse(cmd.Context(), pathURI.Repository, pathURI.Ref, logCommitsParams)
+			DieOnResponseError(res, err)
+			pagination = res.JSON200.Pagination
+			logCommitsParams.After = api.PaginationAfterPtr(pagination.NextOffset)
+			data := struct {
+				Commits    []api.Commit
+				Object     string
+				Pagination *Pagination
+			}{
+				Commits: res.JSON200.Results,
+				Object:  obj.Path,
```

Comment:
I don't understand this code.  This is the *only* place where `obj` is ever used.  It is never passed to the log query.  So AFAIU this will output each object currently under the prefix, and claim that it was last changed at the latest commit.

Perhaps we should move up adding `lakectl` testing to this code, that will probably allow me to understand better.  In the meantime, can you add in a comment some sequences you used for testing?  (Just copy-paste a full session from your terminal...).

Reply:
I have made some changes to the code. The above should be solved.

Reply:
Please let's also do something about testing this code.  Even if for now it's just a statement on the PR of manual steps performed to test it.

Reply:
Will do

Reply:
Please add specific test that were performed -- at the very least, we need to be able to reproduce the tests that you already did.  (Important because clearly it is easy to miss functionality when testing this code...).